folder/src/Component/Copy.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { Button } from "baseui/button";
+
+interface CopyProps {
+  password: string;
+  copied: boolean;
+  setCopied: (copied: boolean) => void;
+}
+
+const Copy: React.FC<CopyProps> = ({ password, setCopied, copied }) => {
+  const handleCopy = () => {
+    navigator.clipboard.writeText(password);
```

Comment:
`writeText` returns a promise, you need to wait for the promise to resolve here.

```tsx
const copyToClipboard = async(): Promise<void> => {
  await navigator.clipboard.writeText(password);
  setCopied(true);
};
```